cw/cw305/pyXKCP/pyXKCP_extention_build.py

```diff
@@ -0,0 +1,95 @@
+#int ParallelHash256_Initialize(ParallelHash_Instance *ParallelHashInstance, size_t blockByteLen, BitLength outputBitLen, const BitSequence *customization, BitLength customBitLen);
+#int ParallelHash256_Update(ParallelHash_Instance *ParallelHashInstance, const BitSequence *input, BitLength inputBitLen);
+#int ParallelHash256_Final(ParallelHash_Instance *ParallelHashInstance, BitSequence * output);
+#int ParallelHash256_Squeeze(ParallelHash_Instance *ParallelHashInstance, BitSequence *output, BitLength outputBitLen);
+#int TupleHash128( const TupleElement *tuple, size_t numberOfElements, BitSequence *output, BitLength outputBitLen, const BitSequence *customization, BitLength customBitLen);
+#int TupleHash128_Initialize(TupleHash_Instance *TupleHashInstance, BitLength outputBitLen, const BitSequence *customization, BitLength customBitLen);
+#int TupleHash128_Update(TupleHash_Instance *TupleHashInstance, const TupleElement *tuple, size_t numberOfElements);
+#int TupleHash128_Final(TupleHash_Instance *TupleHashInstance, BitSequence * output);
+#int TupleHash128_Squeeze(TupleHash_Instance *TupleHashInstance, BitSequence *output, BitLength outputBitLen);
+#int TupleHash256( const TupleElement *tuple, size_t numberOfElements, BitSequence *output, BitLength outputBitLen, const BitSequence *customization, BitLength customBitLen);
+#int TupleHash256_Initialize(TupleHash_Instance *TupleHashInstance, BitLength outputBitLen, const BitSequence *customization, BitLength customBitLen);
+#int TupleHash256_Update(TupleHash_Instance *TupleHashInstance, const TupleElement *tuple, size_t numberOfElements);
+#int TupleHash256_Final(TupleHash_Instance *TupleHashInstance, BitSequence * output);
+#int TupleHash256_Squeeze(TupleHash_Instance *TupleHashInstance, BitSequence *output, BitLength outputBitLen);
+
+ffibuilder.set_source("_XKCP",  # name of the output C extension #ToDo: better naming?
+"""
+    #include "../vendor/xkcp_xkcp/align.h"
+    #include "../vendor/xkcp_xkcp/Phases.h"
+    #include "../vendor/xkcp_xkcp/KeccakSponge.h"
+    #include "../vendor/xkcp_xkcp/SP800-185.h"
+    #include "../vendor/xkcp_xkcp/KeccakP-1600-SnP.h"
+    #include "../vendor/xkcp_xkcp/SnP-Relaned.h"
+    #include "../vendor/xkcp_xkcp/brg_endian.h"
+
+""",
+    sources=['../vendor/xkcp_xkcp/SP800-185.c','../vendor/xkcp_xkcp/KeccakSponge.c','../vendor/xkcp_xkcp/KeccakP-1600-compact64.c'],
+    libraries=[])
+
+
+def create_config_file():
+    with open('../vendor/xkcp_xkcp/config.h', 'w') as the_file:
+#        the_file.write('// Copyright lowRISC contributors\n')
+#        the_file.write('// Licensed under the Apache License, Version 2.0, see LICENSE for details.\n')
+#        the_file.write('// SPDX-License-Identifier: Apache-2.0 \n\n')
+        the_file.write('// This file is autogenerated by ' +  __file__ + '\n// DO NOT CHANGE!\n\n')
+        # Define which KeccakPermutation is used
+        the_file.write('#define XKCP_has_KeccakP1600\n')
+        the_file.close()
+
```

Comment:
You have quite a few line breaks here :-)

What I would recommend is to run `flake8` to check the formatting and style. It will highlight such things. You can also have a look at [our Python style guide](https://docs.opentitan.org/doc/rm/python_coding_style/#lint-tool) on how to run a lint tool on Python. I think this is useful as it will lead to a more consisten style.

Reply:
I'll use the lintpy.py from the opentitan repository as a pre-commit hook.